framework/CMakeLists.txt

```
@@ -450,70 +462,79 @@ endif()
 add_library(${PROJECT_NAME} OBJECT ${PROJECT_FILES})
 set_target_properties(${PROJECT_NAME} PROPERTIES POSITION_INDEPENDENT_CODE ON)
 
+set(PUBLIC_COMPILE_OPTIONS)
+set(PRIVATE_COMPILE_OPTIONS)
+
 # compiler flags based on compiler type
 if(NOT MSVC)
-    target_compile_options(${PROJECT_NAME} PUBLIC -fexceptions)
+    list(APPEND ${PUBLIC_COMPILE_OPTIONS} -fexceptions)
```

When using `list` the second argument should be the name of the variable, not the evaluated contents of the variable... like so...

```suggestion
    list(APPEND PUBLIC_COMPILE_OPTIONS -fexceptions)
```

This applies to everywhere below where you have `${PUBLIC_COMPILE_OPTIONS}` and the like.. these aren't going to do what you think they're going to do.